autonat.go

```diff
@@ -206,14 +216,14 @@ func (as *AmbientAutoNAT) background() {
 		if timerRunning && !timer.Stop() {
 			<-timer.C
 		}
-		timer.Reset(as.scheduleProbe())
+		timer.Reset(as.scheduleProbe(peer))
 		timerRunning = true
 	}
 }
 
 // scheduleProbe calculates when the next probe should be scheduled for,
 // and launches it if that time is now.
-func (as *AmbientAutoNAT) scheduleProbe() time.Duration {
+func (as *AmbientAutoNAT) scheduleProbe(peer peer.ID) time.Duration {
```

Comment:
This function is starting to do way too many things, only one of which is scheduling a probe.